### PR TITLE
Add table oci_database_exadata_infrastructure Closes #570

### LINKED ISSUE #570: Add table oci_database_exadata_infrastructure
**References**
[GetCloudExadataInfrastructure](https://docs.oracle.com/en-us/iaas/api/#/en/database/20160918/CloudExadataInfrastructure/GetCloudExadataInfrastructure)
to obtain the information about an Exadata Infrastructure deployed in OCI
This call will return a single [CloudExadataInfrastructure](https://docs.oracle.com/en-us/iaas/api/#/en/database/20160918/CloudExadataInfrastructure/) resource with all the details about the infrastructure.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A 
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
N/A
```
</details>

### CODE REVIEW

File: oci/table_oci_database_exadata_infrastructure.go [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+				Name:        "maintenance_window",
+				Type:        proto.ColumnType_JSON,
+				Description: "Maintenance Window information.",
+			},
+			{
+				Name:        "contacts",
+				Type:        proto.ColumnType_JSON,
+				Description: "The list of contacts for the Exadata infrastructure.",
+			},
+
```
Comment (madhushreeray30) [inline oci/table_oci_database_exadata_infrastructure.go:255]: We have missed out on adding the following columns:
1. DnsServer
2. NtpServer

File: oci/table_oci_database_exadata_infrastructure.go [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+		CompartmentId: types.String(compartment),
+		Limit:         types.Int(1000),
+		RequestMetadata: common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(d.Connection),
+		},
+	}
+
+	// Check for additional filters
+	if equalQuals["display_name"] != nil {
+		dbName := equalQuals["display_name"].GetStringValue()
```
Comment (madhushreeray30) [inline oci/table_oci_database_exadata_infrastructure.go:325]:
```suggestion
		dbName := d.EqualsQualString("display_name")
```

File: oci/table_oci_database_exadata_infrastructure.go [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+		request.DisplayName = types.String(dbName)
+	}
+
+	if equalQuals["lifecycle_state"] != nil {
+		lifecycleState := equalQuals["lifecycle_state"].GetStringValue()
```
Comment (madhushreeray30) [inline oci/table_oci_database_exadata_infrastructure.go:330]:
```suggestion
		lifecycleState := d.EqualsQualString("lifecycle_state")
```

File: oci/table_oci_database_exadata_infrastructure.go [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+	logger := plugin.Logger(ctx)
+	region := d.EqualsQualString(matrixKeyRegion)
+	compartment := d.EqualsQualString(matrixKeyCompartment)
+
+	// Restrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	id := d.EqualsQuals["id"].GetStringValue()
```
Comment (madhushreeray30) [inline oci/table_oci_database_exadata_infrastructure.go:382]:
```suggestion
	id := d.EqualsQualString("id")
```

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI Please take a look at the comments, thanks!

**Comment by madhushreeray30** [comment]
@ParthaI Please add the example query results, thanks!

**Review by bigdatasourav — APPROVED** [review]

**Pull request merged**